collector/sources.go

```diff
@@ -213,9 +213,17 @@ func (e Exporter) getSourcesMetrics(logger *slog.Logger, ch chan<- prometheus.Me
 			if err != nil {
 				return fmt.Errorf("Failed to get ntpdata response for: %s", r.IPAddr)
 			}
-			ntpData, ok := ntpDataPacket.(*chrony.ReplyNTPData)
-			if !ok {
-				return fmt.Errorf("Got wrong 'ntpdata' response: %q", packet)
+
+			var ntpData *chrony.NTPData
+			replyNTPData, ok := ntpDataPacket.(*chrony.ReplyNTPData)
```

Comment:
I would probably follow the pattern that we do in `parseServerStatsPacket()`.

Something like this:

```suggestion
			var ntpData chrony.NTPData2
			switch stats := p.(type) {
			case *chrony.ReplyNTPData: 
			ntpData = stats
```